pkg/deploy/deploy.go

```diff
@@ -208,51 +210,117 @@ func deployComponentsParallel(ctx context.Context, components []graph.SortedComp
 	return errs
 }
 
-func deployComponent(ctx context.Context, component graph.SortedComponent, clientSet ClientSet, apis api.APIs, opts DeployConfigsOptions) []error {
+type deployGraph interface {
+	graph2.NodeRemover
+	graph2.Directed
+	graph2.Builder
+}
+
+type componentDeployer struct {
+	lock             sync.Mutex
+	graph            deployGraph
+	clients          ClientSet
+	resolvedEntities entityMap
+	apis             api.APIs
+}
+
+func (c *componentDeployer) deployComponent(ctx context.Context) []error {
```

Comment:
nit: why not just `deploy` if it's a function of the componentDeployer already? 

```suggestion
func (c *componentDeployer) deploy(ctx context.Context) []error {
```